album_folder.py

```diff
@@ -4,27 +4,115 @@
 import progressbar
 import os
 import sys
+import time
+from argparse import ArgumentParser
 from shutil import copyfile
+from shutil import move
+
+
+# Source_dir : passed as parameter, where your photos are located
+# output_dir : directory under where all albuns will be created
 
 def run(source_dir, output_dir):
     bar = progressbar.ProgressBar()
-    for f in bar(os.listdir(source_dir)):
-        if os.path.splitext(f)[1] == '.json':
+
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+
+    # Let's print some statistics
+    num_of_missing = 0
+    num_of_albuns = 0
+    num_of_copied = 0
+    num_of_moved = 0
+    num_of_total_files = 0
+    num_of_json = 0
+
+    all_files = os.listdir(source_dir)
+    num_of_total_files = len(all_files)
+
+    for f in bar(all_files):                                       # f is each file in directory
+        (root_file, ext) = os.path.splitext(f)
+        if ext == '.json':                                                      # only .json are treated
+            num_of_json += 1
             with open(os.path.join(source_dir, f)) as data_file:
                 data = json.load(data_file)
-                for album in data['albums']:
-                    album = album.replace("/", "_")
-                    if not os.path.exists(os.path.join(output_dir, album)):
-                        os.mkdir(os.path.join(output_dir, album))
+                album_counter = 1
+
+                file_exported_name = data['uuid']                               # This is the physical exported file
```

Comment:
Please put comments above the line:
```python
# This is the exported file
file_exported_name = data['uuid']
```

Reply:
Ok, but I like to comment as much as possible to easy other people to understand some parts of the code, and I think the comments between lines breaks people to view the core properly.Well... it's just style, somthing personal. Will do anyway.